src/Glpi/Form/Migration/FormMigration.php

```diff
@@ -0,0 +1,488 @@
+            'file'        => QuestionTypeFile::class,
+            'float'       => QuestionTypeNumber::class,
+            'glpiselect'  => QuestionTypeItem::class,
+            'integer'     => QuestionTypeNumber::class,
+            'multiselect' => QuestionTypeDropdown::class,
+            'radios'      => QuestionTypeRadio::class,
+            'requesttype' => QuestionTypeRequestType::class,
+            'select'      => QuestionTypeDropdown::class,
+            'textarea'    => QuestionTypeLongText::class,
+            'text'        => QuestionTypeShortText::class,
+            'time'        => QuestionTypeDateTime::class,
+            'urgency'     => QuestionTypeUrgency::class,
+
+            // Description is replaced by a new block : Comment
+            'description' => null,
+
+            // TODO: Must be implemented
+            'fields'      => null,
+            'tag'         => null,
+
+            // TODO: This types are not supported by the new form system
+            // we need to define alternative ways to handle them
+            'hidden'      => null,
+            'hostname'    => null,
+            'ip'          => null,
+            'ldapselect'  => null,
+            'undefined'   => null,
+        ];
+    }
+
+    protected function validatePrerequisites(): bool
+    {
+        $this->result->addMessage(MessageType::Notice, 'Checking plugin version...');
+
+        $plugin = new Plugin();
+        if (!$plugin->getFromDBbyDir('formcreator')) {
+            $this->result->addMessage(MessageType::Error, 'Formcreator plugin is not installed.');
+            return false;
+        }
+
+        $is_version_ok = self::FORMCREATOR_REQUIRED_VERSION === $plugin->fields['version'];
+        if (!$is_version_ok) {
+            $this->result->addMessage(MessageType::Error, sprintf(
+                'Last Formcreator version (%s) is required to be able to continue.',
+                self::FORMCREATOR_REQUIRED_VERSION
+            ));
+            return false;
+        }
+
+        $formcreator_schema = [
+            'glpi_plugin_formcreator_categories' => [
+                'id', 'name', 'plugin_formcreator_categories_id', 'level'
+            ],
+            'glpi_plugin_formcreator_forms' => [
+                'id', 'name', 'description', 'plugin_formcreator_categories_id', 'entities_id',
+                'is_recursive', 'is_visible'
+            ],
+            'glpi_plugin_formcreator_sections' => [
+                'id', 'name', 'plugin_formcreator_forms_id', 'order', 'uuid'
+            ],
+            'glpi_plugin_formcreator_questions' => [
+                'id', 'name', 'plugin_formcreator_sections_id', 'fieldtype', 'required', 'default_values',
+                'itemtype', 'values', 'description', 'row', 'col', 'uuid'
+            ],
+        ];
+        if (!$this->checkDbFieldsExists($formcreator_schema)) {
+            $this->result->addMessage(MessageType::Error, 'Migration cannot be done.');
```

Comment:
Maybe this could be added directly in the abstract method, when the `validatePrerequisites()` method returns false.